qcfractal/storage_sockets/sql_models.py

```diff
@@ -0,0 +1,557 @@
+import datetime
+# from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import (Column, Integer, String, Text, DateTime, Boolean,
+                        ForeignKey, JSON, Enum, Float, Binary, Table)
+from sqlalchemy.orm import relationship, object_session, column_property
+from qcfractal.interface.models.records import RecordStatusEnum, DriverEnum
+from qcfractal.interface.models.task_models import TaskStatusEnum, ManagerStatusEnum, PriorityEnum
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.ext.declarative import as_declarative
+from sqlalchemy import event, select
+from sqlalchemy.ext.hybrid import hybrid_property
+
+
+# pip install sqlalchemy psycopg2
+
+# Base = declarative_base()
+
+
+@as_declarative()
+class Base:
+    """Base declarative class of all ORM models"""
+
+    def to_dict(self, with_id=True, exclude=None):
+        dict_obj = self.__dict__.copy()
+        del dict_obj['_sa_instance_state']
+        if not with_id:
+            del dict_obj['id']
+        if exclude:
+            for key in exclude:
+                del dict_obj[key]
+        return dict_obj
+
+    @classmethod
+    def col(cls):
+        return cls.__table__.c
+
+    def __str__(self):
+        return str(self.id)
+
+
+class AccessLogORM(Base):
+    __tablename__ = 'access_log'
+
+    id = Column(Integer, primary_key=True)
+    ip_address = Column(String)
+    date = Column(DateTime, default=datetime.datetime.utcnow)
+    type = Column(String)
+
+
+class LogsORM(Base):
+    __tablename__ = "logs"
+
+    id = Column(Integer, primary_key=True)
+    value = Column(Text, nullable=False)
+
+
+class ErrorORM(Base):
+    __tablename__ = "error"
+
+    id = Column(Integer, primary_key=True)
+    value = Column(Text, nullable=False)
+
+
+class CollectionORM(Base):
+    """
+        A collection of precomuted workflows such as datasets, ..
+
+        This is a dynamic document, so it will accept any number of
+        extra fields (expandable and uncontrolled schema)
+    """
+
+    __tablename__ = "collection"
+
+    id = Column(Integer, primary_key=True)
+
+    collection = Column(String(100), nullable=False)
+    name = Column(String(100), nullable=False)  # Example 'water'
+
+    tags = Column(JSON)
```

Comment:
I agree. But lists are not supported in SQLite. I think we should drop support for SQLite since it's going to be very restrictive to the more general case.

Reply:
It would be good to try to stick the having a base socket that can connect via SQLite for the Snowflake case. If it becomes too hard so be it, but good to try.

Reply:
I'm trying my best to keep it compatible with SQLite, but I kept finding limitations with SQLite. For instance, many times we set dates as strings, coming from Pydantic serialization. Postgres, MySQL or mongodb can convert that automatically, but to be compatible with the simple SQLite, we need to convert every created_on or modified_on to be Python dates explicitly not str. 
We won't be able to use lists, which we can index for searching tags. This is not compatible with SQLite which doesn't support lists.